templates/nextjs/package.json

```diff
@@ -9,7 +9,9 @@
     "build": "pnpm run xprebuild && next build",
     "start": "next start",
     "lint": "next lint",
-    "test": "vitest"
+    "test": "run-s test:*",
+    "test:e2e": "vitest",
```

Comment:
This is just a terminology thing, but are these end-to-end tests?

Reply:
No, that specific script is for the e2e tests that we already have. For the tests added in this PR, the command is `test:unit`

Reply:
Yes, I was referring to the existent tets (`test` -> `test:e2e`).

Are these actual end-to-end tests?

Reply:
Well, it depends on what you classify as end-to-end. 

These tests do cover starting a node, deploying a contract instance, calling that contract and assserting the result. 

But they don't cover the UI part.

Reply:
Given how we currently use these terms, I guess it'd be an integration test.

@danielbate Thoughts?

Reply:
Yah agreed let's stick with our internal classification.

```jsonc
{
  "test": "run-s test:*",
  "test:unit": "forc test",
  "test:integration": "vitest ..", // launchTestNode
  "test:ui": "vitest ..", // playwright
}
```

`e2e` would be against a live node.

Reply:
I'd consider removing the serial test and letting `vitest` run off `test` as I think that will be more common practice, like we do internally.

Reply:
Renamed scripts here and updated folders/scripts in https://github.com/FuelLabs/fuels-ts/pull/3104 too

Reply:
@Dhaiwat10 @danielbate Another [slightly shorter] term could be `feat`_ures_. ([example](https://github.com/FuelLabs/fuels-ts/tree/master/packages/fuels/test/features))

Regarding the commands, what if we suggest a default and provide a **_parallel_** `run all` option?

In this case, more options may also provide workflow inspiration, even for CI.

```json
{
  "test": "run-p test:feat",
  "test:all": "run-p test:feat test:unit test:ui",
  "test:feat": "vitest",
  "test:unit": "forc test",
  "test:ui": "sh ./scripts/test-ui.sh.",
}
```

Reply:
`feat` sounds good to me.

Like the idea of suggesting a default too. Makes the most sense.

Reply:
https://github.com/FuelLabs/fuels-ts/pull/3100/commits/3dedab6ac535909a6812cfb8291d17fb893b3691